In [32]:
import pandas as pd
import numpy as np
import warnings
warnings.filterwarnings('ignore')

# Display settings used by professionals
pd.set_option('display.max_columns', 20)
pd.set_option('display.max_rows', 30)
pd.set_option('display.float_format', '{:.2f}'.format)

print(f"pandas {pd.__version__}  |  numpy {np.__version__}  ✅")


pandas 3.0.3  |  numpy 2.4.6  ✅


In [33]:
df_immovlan = pd.read_csv("../data/raw/properties_cleaned.csv")
print(type(df_immovlan))

<class 'pandas.DataFrame'>


In [34]:
# Shape
print("Shape:", df_immovlan.shape)
print(f"{df_immovlan.shape[0]:,} rows × {df_immovlan.shape[1]} columns\n")

Shape: (11501, 34)
11,501 rows × 34 columns



In [35]:
# dtypes + non-null counts
df_immovlan.info()

<class 'pandas.DataFrame'>
RangeIndex: 11501 entries, 0 to 11500
Data columns (total 34 columns):
 #   Column                      Non-Null Count  Dtype  
---  ------                      --------------  -----  
 0   property_id                 11501 non-null  str    
 1   property_type               11501 non-null  str    
 2   property_subtype            11501 non-null  str    
 3   price                       11292 non-null  float64
 4   price_type                  11501 non-null  str    
 5   living_area_m2              10707 non-null  float64
 6   bedrooms                    11090 non-null  float64
 7   bathrooms                   10203 non-null  float64
 8   address                     9299 non-null   str    
 9   postal_code                 11501 non-null  int64  
 10  city                        11501 non-null  str    
 11  latitude                    11499 non-null  float64
 12  longitude                   11499 non-null  float64
 13  building_year               6790 non-null 

In [36]:
# First & last 3 rows
print("--- HEAD ---")
display(df_immovlan.head(3))
print("\n--- TAIL ---")
display(df_immovlan.tail(3))

--- HEAD ---


,property_id,property_type,property_subtype,price,price_type,living_area_m2,bedrooms,bathrooms,address,postal_code,...,has_terrace,total_area_m2,epc_score,region,province,nearby_city,km_from_nearby_city,is_nearby_city_prestigious,floor_number,property_url
0,https://immovlan.be/en/detail/residence/for-sa...,House,residence,550000.00,sell,250.00,4.00,1.00,Rue des Nonnes 32A,4400,...,1.00,780.00,A+,Wallonia,Liège,Liège,11.80,0.00,NaN,https://immovlan.be/en/detail/residence/for-sa...
1,https://immovlan.be/en/detail/apartment/for-sa...,Apartment,apartment,200000.00,sell,57.00,1.00,1.00,Avenue Louise 32A,1050,...,1.00,NaN,B,Brussels,Brussels Capital Region,Bruxelles,1.80,0.00,2.00,https://immovlan.be/en/detail/apartment/for-sa...
2,https://immovlan.be/en/detail/duplex/for-sale/...,Apartment,duplex,189000.00,sell,115.00,1.00,1.00,NaN,7500,...,1.00,NaN,G,Wallonia,Hainaut,Tournai,0.10,0.00,NaN,https://immovlan.be/en/detail/duplex/for-sale/...



--- TAIL ---


,property_id,property_type,property_subtype,price,price_type,living_area_m2,bedrooms,bathrooms,address,postal_code,...,has_terrace,total_area_m2,epc_score,region,province,nearby_city,km_from_nearby_city,is_nearby_city_prestigious,floor_number,property_url
11498,https://immovlan.be/en/detail/residence/for-sa...,House,residence,320000.00,sell,220.00,5.00,NaN,Rue des Quatre Arbres 11,5170,...,0.00,856.00,D,Wallonia,Namur,Namur,10.10,0.00,NaN,https://immovlan.be/en/detail/residence/for-sa...
11499,https://immovlan.be/en/detail/studio/for-sale/...,Apartment,studio,153000.00,sell,13.00,NaN,1.00,Rue André Masquelier 11,7000,...,NaN,NaN,NaN,Wallonia,Hainaut,Mons,0.70,0.00,0.00,https://immovlan.be/en/detail/studio/for-sale/...
11500,https://immovlan.be/en/detail/residence/for-sa...,House,residence,535000.00,sell,177.00,4.00,2.00,Spijkstraat 91-93,9000,...,1.00,155.00,NaN,Flanders,East Flanders,Gent,1.90,0.00,NaN,https://immovlan.be/en/detail/residence/for-sa...


In [37]:
# Summary statistics
df_immovlan.describe(include='all')

,property_id,property_type,property_subtype,price,price_type,living_area_m2,bedrooms,bathrooms,address,postal_code,...,has_terrace,total_area_m2,epc_score,region,province,nearby_city,km_from_nearby_city,is_nearby_city_prestigious,floor_number,property_url
count,11501,11501,11501,11292.00,11501,10707.00,11090.00,10203.00,9299,11501.00,...,9075.00,6352.00,9623,11501,11501,11499,11499.00,11499.00,3312.00,11501
unique,11500,2,15,NaN,1,NaN,NaN,NaN,8744,NaN,...,NaN,NaN,12,3,9,42,NaN,NaN,NaN,11500
top,https://immovlan.be/en/detail/residence/for-sa...,House,residence,NaN,sell,NaN,NaN,NaN,Dessauerplein 1,NaN,...,NaN,NaN,C,Wallonia,Liège,Liège,NaN,NaN,NaN,https://immovlan.be/en/detail/residence/for-sa...
freq,2,7406,6015,NaN,11501,NaN,NaN,NaN,31,NaN,...,NaN,NaN,1872,5582,1734,1299,NaN,NaN,NaN,2
mean,NaN,NaN,NaN,388215.93,NaN,173.37,3.07,1.40,NaN,5269.59,...,0.82,1646.79,NaN,NaN,NaN,NaN,37.20,0.05,3.61,NaN
std,NaN,NaN,NaN,357484.06,NaN,160.29,2.18,0.98,NaN,2621.05,...,0.38,5464.68,NaN,NaN,NaN,NaN,403.64,0.22,30.11,NaN
min,NaN,NaN,NaN,1.00,NaN,1.00,1.00,1.00,NaN,1000.00,...,0.00,2.00,NaN,NaN,NaN,NaN,0.00,0.00,0.00,NaN
25%,NaN,NaN,NaN,190000.00,NaN,90.00,2.00,1.00,NaN,3500.00,...,1.00,260.00,NaN,NaN,NaN,NaN,3.10,0.00,1.00,NaN
50%,NaN,NaN,NaN,299000.00,NaN,143.00,3.00,1.00,NaN,5190.00,...,1.00,626.00,NaN,NaN,NaN,NaN,8.30,0.00,2.00,NaN
75%,NaN,NaN,NaN,495000.00,NaN,206.00,4.00,2.00,NaN,7370.00,...,1.00,1338.50,NaN,NaN,NaN,NaN,16.90,0.00,3.00,NaN


**copy the dataset before applying any change**

In [148]:
df_immo = df_immovlan.copy()

In [149]:
columns_to_control = ['living_area_m2', 'total_area_m2','price']
suspect_vals_cols = ['bathrooms', 'bedrooms', 'floor_number']
for suspect in suspect_vals_cols:
    if suspect == 'bathrooms':
        cond = (df_immo['bathrooms'] > 5.00)
    elif suspect == 'bedrooms':
        cond = (df_immo['bedrooms'] > 5.00)
    else:
        cond = (df_immo['floor_number'] == 999.00)

    print(f"Property url: {df_immo.loc[cond, 'property_url']}")

Property url: 109      https://immovlan.be/en/detail/residence/for-sa...
142      https://immovlan.be/en/detail/mixed-building/f...
157      https://immovlan.be/en/detail/mixed-building/f...
163      https://immovlan.be/en/detail/residence/for-sa...
194      https://immovlan.be/en/detail/residence/for-sa...
                               ...                        
11388    https://immovlan.be/en/detail/mixed-building/f...
11415    https://immovlan.be/en/detail/residence/for-sa...
11420    https://immovlan.be/en/detail/residence/for-sa...
11439    https://immovlan.be/en/detail/residence/for-sa...
11474    https://immovlan.be/en/detail/residence/for-sa...
Name: property_url, Length: 86, dtype: str
Property url: 7        https://immovlan.be/en/detail/mixed-building/f...
15       https://immovlan.be/en/detail/residence/for-sa...
55       https://immovlan.be/en/detail/residence/for-sa...
86       https://immovlan.be/en/detail/residence/for-sa...
90       https://immovlan.be/en/detail/resid

In [150]:
# printing temporary df

condition = (df_immo['bathrooms'] > 5.00)
df_immo_target = df_immo[condition].sort_values(by='bathrooms', ascending=False)

indexes_to_be_deleted = df_immo_target[df_immo_target['price'].isna()].index.tolist()
print(indexes_to_be_deleted)

df_immo_to_print = df_immo_target[df_immo_target['price'].notna()]

df_immo_to_print.loc[condition, ['bathrooms'] + columns_to_control]

[8004, 4813, 2188, 9806, 7690]


,bathrooms,living_area_m2,total_area_m2,price
604,23.00,1280.00,10808.00,990000.00
8303,22.00,1498.00,NaN,4950000.00
9648,15.00,560.00,1084.00,340000.00
3932,14.00,NaN,NaN,750000.00
9111,14.00,600.00,11050.00,2300000.00
...,...,...,...,...
9493,6.00,204.00,NaN,475000.00
10721,6.00,230.00,1025.00,360000.00
10943,6.00,558.00,1506.00,450000.00
11415,6.00,317.00,166.00,575000.00


In [151]:

# printing values in columns_to_control for each row and deleting rows with NaN price

for index, row in df_immo_to_print.iterrows():
    print(f"property with {row['bathrooms']} in 'bathrooms' column (Index: {index}) has:\n")
    for column in columns_to_control:
        print(f"  {row[column]} in {column}.\n")

# deletion of found rows with NaN price value
df_immo.drop(indexes_to_be_deleted)
deleted_indexes = ", ".join(map(str, indexes_to_be_deleted))
print(f'Rows with labels {deleted_indexes} with NaN price habe been successfully deleted.')

property with 23.0 in 'bathrooms' column (Index: 604) has:

  1280.0 in living_area_m2.

  10808.0 in total_area_m2.

  990000.0 in price.

property with 22.0 in 'bathrooms' column (Index: 8303) has:

  1498.0 in living_area_m2.

  nan in total_area_m2.

  4950000.0 in price.

property with 15.0 in 'bathrooms' column (Index: 9648) has:

  560.0 in living_area_m2.

  1084.0 in total_area_m2.

  340000.0 in price.

property with 14.0 in 'bathrooms' column (Index: 3932) has:

  nan in living_area_m2.

  nan in total_area_m2.

  750000.0 in price.

property with 14.0 in 'bathrooms' column (Index: 9111) has:

  600.0 in living_area_m2.

  11050.0 in total_area_m2.

  2300000.0 in price.

property with 14.0 in 'bathrooms' column (Index: 5774) has:

  1100.0 in living_area_m2.

  1172.0 in total_area_m2.

  550000.0 in price.

property with 14.0 in 'bathrooms' column (Index: 11388) has:

  391.0 in living_area_m2.

  459.0 in total_area_m2.

  790000.0 in price.

property with 12.0 in 'bathro

*deletion of row 604 to be considered*

In [152]:
print(f"property with {df_immo['bedrooms'].max()} in 'bedrooms' column:")
df_immo.loc[(df_immo['bedrooms'] == 100.00), ['living_area_m2', 'total_area_m2', 'bedrooms','price']]

property with 100.0 in 'bedrooms' column:


,living_area_m2,total_area_m2,bedrooms,price
5214,NaN,NaN,100.00,13000000.00


**deliting row 5214**

In [153]:
to_be_deleted = [5214]
deleted_indexes = ", ".join(map(str, to_be_deleted))
df_immo = df_immo.drop(to_be_deleted)
print(f'Rows with labels {deleted_indexes} successfully deleted.')

Rows with labels 5214 successfully deleted.


**assessing other rows situation that have high value of bedrooms**

In [154]:
# printing temporary df

condition = (df_immo['bedrooms'] > 5.00)
df_immo_target = df_immo[condition].sort_values(by='bedrooms', ascending=False)

indexes_to_be_deleted = df_immo_target[df_immo_target['price'].isna()].index.tolist()
print(indexes_to_be_deleted)

df_immo_to_print = df_immo_target[df_immo_target['price'].notna()]

df_immo_to_print.loc[condition, ['bedrooms'] + columns_to_control]

[4813, 8004, 3831, 11276, 2188, 5613, 2176, 5818, 8192, 5059, 2941, 7690, 2221, 9806, 928, 5519, 6141, 6580, 3523, 3305, 9236, 9154]


,bedrooms,living_area_m2,total_area_m2,price
3081,63.00,2378.00,1232.00,650000.00
6056,50.00,3990.00,2495.00,1988750.00
1556,50.00,3500.00,33704.00,1400000.00
2797,41.00,NaN,6000.00,1499000.00
586,32.00,840.00,1017.00,795000.00
...,...,...,...,...
11049,6.00,299.00,300.00,545000.00
11034,6.00,369.00,521.00,549950.00
11032,6.00,280.00,408.00,525000.00
11008,6.00,294.00,423.00,465000.00


In [155]:
# printing values in columns_to_control for each row and deleting rows with NaN price

for index, row in df_immo_to_print.iterrows():
    print(f"property with {row['bedrooms']} in 'bedrooms' column (Index: {index}) has:\n")
    for column in columns_to_control:
        print(f"  {row[column]} in {column}.\n")

# deletion of found rows with NaN price value
df_immo.drop(indexes_to_be_deleted)
deleted_indexes = ", ".join(map(str, indexes_to_be_deleted))
print(f'Rows with labels {deleted_indexes} with NaN price habe been successfully deleted.')

property with 63.0 in 'bedrooms' column (Index: 3081) has:

  2378.0 in living_area_m2.

  1232.0 in total_area_m2.

  650000.0 in price.

property with 50.0 in 'bedrooms' column (Index: 6056) has:

  3990.0 in living_area_m2.

  2495.0 in total_area_m2.

  1988750.0 in price.

property with 50.0 in 'bedrooms' column (Index: 1556) has:

  3500.0 in living_area_m2.

  33704.0 in total_area_m2.

  1400000.0 in price.

property with 41.0 in 'bedrooms' column (Index: 2797) has:

  nan in living_area_m2.

  6000.0 in total_area_m2.

  1499000.0 in price.

property with 32.0 in 'bedrooms' column (Index: 586) has:

  840.0 in living_area_m2.

  1017.0 in total_area_m2.

  795000.0 in price.

property with 30.0 in 'bedrooms' column (Index: 358) has:

  3000.0 in living_area_m2.

  14229.0 in total_area_m2.

  1350000.0 in price.

property with 30.0 in 'bedrooms' column (Index: 8876) has:

  1380.0 in living_area_m2.

  6149.0 in total_area_m2.

  950000.0 in price.

property with 26.0 in 'bedr

In [157]:
# floor_number outlier
print(f"property with {df_immo['floor_number'].max()} in {'floor_number'} column:")
condition = (df_immo['floor_number'] == 999.00)
df_immo_target = df_immo[condition]
indexes_to_be_manipulated = df_immo_target.index.tolist()

df_immo.loc[condition, ['floor_number'] + columns_to_control]

property with 999.0 in floor_number column:


,floor_number,living_area_m2,total_area_m2,price
735,999.00,176.00,NaN,979000.00
3314,999.00,84.00,NaN,159000.00
7262,999.00,NaN,NaN,279000.00


**rows 735, 3314 and 7262 contain valuable infos - I sobstitute the floor number with NaN**

In [159]:
# converting 999 to NaN

for row in indexes_to_be_manipulated:
    df_immo.loc[condition, 'floor_number'] = np.nan
print(f'Floor number in rows labels {indexes_to_be_manipulated} successfully substituted with NaN.')

Floor number in rows labels [735, 3314, 7262] successfully substituted with NaN.


**delete ALL THE OTHER ROWS with nan price**

In [161]:
indexes_to_be_deleted = df_immo[df_immo['price'].isna()].index.tolist()

df_immo.drop(indexes_to_be_deleted)
deleted_indexes = ", ".join(map(str, indexes_to_be_deleted))
print(f'Rows with labels {deleted_indexes} with NaN price habe been successfully deleted.')

Rows with labels 152, 160, 369, 397, 400, 493, 562, 673, 686, 692, 824, 847, 848, 873, 928, 1017, 1102, 1116, 1156, 1206, 1240, 1280, 1281, 1282, 1377, 1592, 1644, 1691, 1793, 1796, 1840, 1859, 1925, 1960, 1964, 2015, 2075, 2176, 2188, 2198, 2221, 2308, 2311, 2629, 2680, 2686, 2695, 2735, 2773, 2808, 2820, 2864, 2901, 2921, 2941, 2974, 3011, 3028, 3064, 3071, 3125, 3128, 3150, 3164, 3166, 3181, 3292, 3304, 3305, 3467, 3523, 3549, 3563, 3569, 3612, 3631, 3637, 3709, 3718, 3785, 3831, 3959, 3961, 4040, 4256, 4258, 4347, 4350, 4360, 4377, 4417, 4612, 4627, 4744, 4780, 4813, 4850, 4856, 4930, 5044, 5059, 5090, 5170, 5328, 5374, 5443, 5509, 5515, 5519, 5552, 5613, 5779, 5801, 5803, 5818, 5911, 5929, 5959, 6073, 6141, 6164, 6173, 6249, 6505, 6554, 6580, 6602, 6606, 6750, 6790, 6796, 6869, 6995, 7164, 7171, 7179, 7255, 7337, 7346, 7418, 7421, 7513, 7567, 7632, 7683, 7690, 7750, 7977, 8004, 8013, 8043, 8108, 8161, 8192, 8375, 8474, 8490, 8601, 8634, 8664, 8713, 8773, 8889, 8890, 8924, 8955, 89

**look at describe again**

In [162]:
df_immo.describe(include='all')

,property_id,property_type,property_subtype,price,price_type,living_area_m2,bedrooms,bathrooms,address,postal_code,...,has_terrace,total_area_m2,epc_score,region,province,nearby_city,km_from_nearby_city,is_nearby_city_prestigious,floor_number,property_url
count,11500,11500,11500,11291.00,11500,10707.00,11089.00,10203.00,9299,11500.00,...,9075.00,6352.00,9623,11500,11500,11498,11498.00,11498.00,3309.00,11500
unique,11499,2,15,NaN,1,NaN,NaN,NaN,8744,NaN,...,NaN,NaN,12,3,9,42,NaN,NaN,NaN,11499
top,https://immovlan.be/en/detail/residence/for-sa...,House,residence,NaN,sell,NaN,NaN,NaN,Dessauerplein 1,NaN,...,NaN,NaN,C,Wallonia,Liège,Liège,NaN,NaN,NaN,https://immovlan.be/en/detail/residence/for-sa...
freq,2,7405,6014,NaN,11500,NaN,NaN,NaN,31,NaN,...,NaN,NaN,1872,5582,1734,1299,NaN,NaN,NaN,2
mean,NaN,NaN,NaN,387098.96,NaN,173.37,3.07,1.40,NaN,5269.97,...,0.82,1646.79,NaN,NaN,NaN,NaN,37.20,0.05,2.71,NaN
std,NaN,NaN,NaN,337218.93,NaN,160.29,1.97,0.98,NaN,2620.87,...,0.38,5464.68,NaN,NaN,NaN,NaN,403.66,0.22,2.81,NaN
min,NaN,NaN,NaN,1.00,NaN,1.00,1.00,1.00,NaN,1000.00,...,0.00,2.00,NaN,NaN,NaN,NaN,0.00,0.00,0.00,NaN
25%,NaN,NaN,NaN,190000.00,NaN,90.00,2.00,1.00,NaN,3500.00,...,1.00,260.00,NaN,NaN,NaN,NaN,3.10,0.00,1.00,NaN
50%,NaN,NaN,NaN,299000.00,NaN,143.00,3.00,1.00,NaN,5190.00,...,1.00,626.00,NaN,NaN,NaN,NaN,8.30,0.00,2.00,NaN
75%,NaN,NaN,NaN,495000.00,NaN,206.00,4.00,2.00,NaN,7370.00,...,1.00,1338.50,NaN,NaN,NaN,NaN,16.90,0.00,3.00,NaN


**do the same exploratory work on rows which have the lowest price (ascending=True)**

**-------------------------------------------------------------------------------------------------------------**

In [27]:
# Missing values map
missing = df_immovlan.isnull().sum()
pct = (missing / len(df_immovlan) * 100).round(1)
pd.DataFrame({'count': missing, 'pct%': pct}).query('count > 0')

,count,pct%
price,209,1.80
living_area_m2,794,6.90
bedrooms,411,3.60
bathrooms,1298,11.30
address,2202,19.10
latitude,2,0.00
longitude,2,0.00
building_year,4711,41.00
state_of_the_building,2682,23.30
furnished,4689,40.80


In [28]:
# Cardinality of categorical columns
for col in ['property_type', 'price', 'region']:
    print(f"{col}: {df_immovlan[col].nunique()} unique → {df_immovlan[col].unique()}")

property_type: 2 unique → <StringArray>
['House', 'Apartment']
Length: 2, dtype: str
price: 1763 unique → [550000. 200000. 189000. ... 692774. 390744. 448492.]
region: 3 unique → <StringArray>
['Wallonia', 'Brussels', 'Flanders']
Length: 3, dtype: str


**Checking data types**

In [29]:
# Data type and duplicates
print("Data types at the start:")
print(df_immovlan.dtypes)

Data types at the start:
property_id                       str
property_type                     str
property_subtype                  str
price                         float64
price_type                        str
                               ...   
nearby_city                       str
km_from_nearby_city           float64
is_nearby_city_prestigious    float64
floor_number                  float64
property_url                      str
Length: 34, dtype: object


**Setting categories and checking again**

In [30]:
print(df_immovlan['region'].unique())

<StringArray>
['Wallonia', 'Brussels', 'Flanders']
Length: 3, dtype: str


In [31]:
df_immovlan['region'] = pd.Categorical(
    df_immovlan['region'],
    categories=df_immovlan['region'].unique(),
    ordered=False)

df_immovlan['property_type'] = pd.Categorical(
    df_immovlan['property_type'],
    categories=df_immovlan['property_type'].unique(),
    ordered=False)

df_immovlan['property_subtype'] = pd.Categorical(
    df_immovlan['property_subtype'],
    categories=df_immovlan['property_subtype'].unique(),
    ordered=False)

print("Data types after categorizing:")
print(df_immovlan.dtypes)

Data types after categorizing:
property_id                        str
property_type                 category
property_subtype              category
price                          float64
price_type                         str
                                ...   
nearby_city                        str
km_from_nearby_city            float64
is_nearby_city_prestigious     float64
floor_number                   float64
property_url                       str
Length: 34, dtype: object


In [ ]:
# **Checking for duplicates**

In [66]:
print("Number of duplicate properties:")
print(df_immovlan.duplicated().sum())

Number of duplicate properties:
0


# **Investigate NaN data**

In [61]:
df_immov = df_immovlan.copy()

# printing proportion of missing data in decreasing order of percentage
# print((df_immov.isna().mean()* 100).sort_values(ascending=False))

# printing the columns names with highest values of nan %
filtrated_col = (
    df_immov.isna().mean()* 100).sort_values(ascending=False)[
    (df_immov.isna().mean()* 100).sort_values(ascending=False) > 2]
print("Columns with nan percentage higher than 2:")
print(filtrated_col)



Columns with nan percentage higher than 2:
garden_area_m2          78.11
kitchen_equipped        73.84
floor_number            71.20
parking_count           66.99
has_garage              56.47
floors_total            55.27
total_area_m2           44.77
building_year           40.96
furnished               40.77
has_elevator            29.05
has_garden              27.95
facades                 25.10
state_of_the_building   23.32
has_terrace             21.09
address                 19.15
epc_score               16.33
bathrooms               11.29
living_area_m2           6.90
bedrooms                 3.57
dtype: float64


In [65]:
# grouping data by "property_type" and checking again for nan %
grouped_nan_perc = df_immov['floor_number'].isna().groupby(df_immov['property_type']).mean() * 100
print(grouped_nan_perc)
print("-" * 30)
print("\n")
print(f'⚠️ "Apartment" type has {grouped_nan_perc.loc['Apartment'].round(2)} percent nan.')
print("Fill 'floor_number' with median or mode.")
print("\n")
print(f'⚠️ "House" type has {grouped_nan_perc.loc['Apartment'].round(2)} percent nan.')
print("No 'floor_number' data for this group.")

property_type
House       100.00
Apartment    19.12
Name: floor_number, dtype: float64
------------------------------


⚠️ "Apartment" type has 19.12 percent nan.
Fill 'floor_number' with median or mode.


⚠️ "House" type has 19.12 percent nan.
No 'floor_number' data for this group.


In [ ]:
# grouping data by "property_type" and checking again for nan %
grouped_nan_perc = df_immov['floor_number'].isna().groupby(df_immov['property_type']).mean() * 100
print(grouped_nan_perc)
print("-" * 30)
print("\n")
print(f'⚠️ "Apartment" type has {grouped_nan_perc.loc['Apartment'].round(2)} percent nan.')
print("Fill 'floor_number' with median ore mode.")
print("\n")
print(f'⚠️ "House" type has {grouped_nan_perc.loc['Apartment'].round(2)} percent nan.')
print("No 'floor_number' data for this group.")

# **Standardize string data**

In [38]:
# isolating string type columns
string_columns = df_immovlan.select_dtypes(include=['object'])

# go through these columns to check for unique values
for column in string_columns:
    print(f'Unique values in column {column}:')
    print(df_immovlan[column].unique())
    print("-" * 30)

Unique values in column property_id:
<StringArray>
[               'https://immovlan.be/en/detail/residence/for-sale/4400/flemalle/vbd65018',
                  'https://immovlan.be/en/detail/apartment/for-sale/1050/elsene/vbe35215',
                    'https://immovlan.be/en/detail/duplex/for-sale/7500/tournai/vbe33657',
                   'https://immovlan.be/en/detail/residence/for-sale/3941/eksel/rbv74485',
     'https://immovlan.be/en/detail/residence/for-sale/6980/la-roche-en-ardenne/vbe31485',
  'https://immovlan.be/en/detail/residence/for-sale/1200/sint-lambrechts-woluwe/vbe35271',
 'https://immovlan.be/en/detail/residence/for-sale/7160/chapelle-lez-herlaimont/vbd97558',
              'https://immovlan.be/en/detail/mixed-building/for-sale/5000/namur/vbc95048',
                   'https://immovlan.be/en/detail/bungalow/for-sale/6940/durbuy/vbd83323',
                  'https://immovlan.be/en/detail/villa/for-sale/3850/binderveld/rbw10337',
 ...
   'https://immovlan.be/en/detail/

***must check for blank spaces***